File: src/main/java/uk/yermak/audiobookconverter/MediaInfo.java
package uk.yermak.audiobookconverter;

public interface MediaInfo extends Organisable {

    int getChannels();

    int getFrequency();

    int getBitrate();

    String getFileName();

    AudioBookInfo getBookInfo();

    ArtWork getArtWork();

    String getCodec();

    long getOffset();

    void setDuration(final long duration);

    void setChapter(Chapter chapter);

    public Chapter getChapter();

    public int getUID();
}

        

File: src/main/scala/uk/yermak/audiobookconverter/MediaInfo.scala
package uk.yermak.audiobookconverter

/**
  * Created by Yermak on 03-Jan-18.
  */
trait MediaInfo {
  def setChannels(channels: Int): Unit

  def setFrequency(frequency: Int): Unit

  def setBitrate(bitrate: Int): Unit

  def setDuration(duration: Long): Unit

  def getChannels: Int

  def getFrequency: Int

  def getBitrate: Int

  def getDuration: Long

  def getFileName: String

  def setBookInfo(bookInfo: AudioBookInfo): Unit

  def getBookInfo: AudioBookInfo

  def getArtWork: ArtWork

  def setArtWork(artWork: ArtWork): Unit
}
